server/src/main/java/org/elasticsearch/action/bulk/BulkRequestParser.java

```diff
@@ -299,4 +297,33 @@ public void parse(
         }
     }
 
+    private static XContentParser createParser(BytesReference data, XContent xContent) throws IOException {
+        if (data instanceof BytesArray) {
+            return parseBytesArray(xContent, (BytesArray) data, 0, data.length());
+        } else {
+            return xContent.createParser(NamedXContentRegistry.EMPTY, LoggingDeprecationHandler.INSTANCE, data.streamInput());
```

Comment:
Aha, here is where the `EMPTY` went. You could re-insert the comment about it here.